### DISCUSSION TITLE
Using wagmi v1.0.5 how to get signer?

### DISCUSSION THREAD

**Discussion by ColeMiller21** [wevm/wagmi#2422] [Q&A]
I am building an application to go on mainnet and using bscTestnet to test. I need to get the signer to put as the third argument for the ethers Contract. I realized that useProvider and useSigner have been replaced with WalletClient and PublicClient but inputing those do not work either? 

I was doing this - new ethers.providers.Web3Provider(connector?.options.getProvider())
But that only works for metamask when I need to get the signer no matter what wallet is connected (Right now just using the default wallets)

if I use the WalletClient I get account unknown #0 error?

Any pointers?? I feel like I am missing something obvious but have been trying to figure it out with no luck! So hopefully someone can help or point me in the right direction.

**Comment by tmm** [comment]
Might be some helpful info here: https://github.com/wagmi-dev/wagmi/discussions/2340
